### PR TITLE
Add a Composer script for autofixing PHP code style issues

### PR DESCRIPTION
**Pull request by oliverklee**

### CODE REVIEW

**Comment by Danon** [comment]
I'm really not sure about that.

Won't take make contributors run this script, and then not pay attention to whatever the script changed, and just blindly commit?

**Comment by oliverklee** [comment]
> Won't take make contributors run this script, and then not pay attention to whatever the script changed, and just blindly commit?

I don't see this as a real problem. Autofixing style issues is a real time saver, and independent of how someone makes some changes, they still need to read their code before (or when) they create a PR for these changes.

(Apart from that, in my experience `phpcbf` so far never has changed the code in a way that was a problem.)

**Comment by Danon** [comment]
I really have a mixed feelings about those "autofixed" solutions.

First of all, why is it even named "fixed"? Is there something broken, that requires fixing? I don't think so. What it does, is conforming the current codebase to the arbitrarily set rules. Shouldn't it be named `conform-to-codestyle-rules` then? Certainly not "fix"/"autofix". I hope you know what I mean? (I don't mean to be too bold).

Second of all, are you sure we should empose rules on the code, that are so hard to maintain that you require a script to follow them for you? 

I mean, it's hard to follow the intention here. The point of the code-style/code-check is to find potential problems/bugs that humans failed to see. That's right, right? And then... we introduce a tool, that can be used by programmers to bypass the very same tool we introduced before, to "autofix" it. The tool of code-style/code-check, at least I respected, because it would allow me to find the mistakes I made. I could use it. It could help me make the library better. But the "autofixed" is there to fool me. It will hide any mistakes I could've done from me and edit the codebase according to the style. If there was indeed a real bug or a real mistake, I will never find it.

I'm sorry, perhaps I'm missing some key-features. Maybe you can help me see it? Maybe, as you're the more experienced user of these tools can help me see the overall picture?

But the way I see it, I have a choice:
 - 1% chance of missing a bug, that the autofixed will hide from me
 - 99% chance of inconvenience, annoyance, and time spent manually adapting code according to the code-style

for me it isn't a chocie at all. 

PS: This `autofixer` idea in my head is very similar to the idea of `autotester`, which would automatically write a unit test for a logic, to match the coverage. And that's of course none-sense.

**Comment by oliverklee** [comment]
Then let's close this PR. (I don't really want to spend so much time discussing this. If you'd like me to explain my reasoning and how you might have misunderstood some things, please let me know. Maybe a call would also work better then communicating via GitHub issues. ;-) )

**Comment by oliverklee** [comment]
(I'd be happy to explain this to you, but only if you're interested in hearing it.)

**Comment by Danon** [comment]
@oliverklee Sure, I'd be interested in hearing the explaination.

Things that don't quite add up to me:

If using `autofixer` is good and safe, why do we need the `checker` and why do humans should even know the rules. surely, if the `autofixer` can be relied on, then we can trust it to introduce any kind of change the autofixer is programmered to do. In that case, checkers and rules should be out of our control. However, if you don't find that true, that autoprefixes isn't 100% reliable, then why use it. And if it's not reliable, then humans would have to find occurrances of that unreliablness.

**Comment by oliverklee** [comment]
Okay. Should we have a call?

I propose we chat in order to find a good date and time. I'm at the Symfony Devs and the TYPO3 Slack.

**Comment by Danon** [comment]
@oliverklee I created `danon.t-regx` user in TYPO3. I think I need an invitation to join the slack.

**Comment by oliverklee** [comment]
@Danon Now that you have a typo3.org account, you can log in at https://my.typo3.org/ and navigate to https://my.typo3.org/profile . Under the heading "Connected services", there should be an option to connect your account to the TYPO3 Slack workspace.

**Comment by oliverklee** [comment]
@Danon Please let's coordinate a good date and time for a call via Slack.
